v3/src/data-interactive/handlers/color-map-handler.ts

```diff
@@ -0,0 +1,87 @@
+import { createProvisionalCategorySet } from "../../models/data/category-set"
+import { getSharedCaseMetadataFromDataset } from "../../models/shared/shared-data-utils"
+import { DICategoryColorMap } from "../data-interactive-data-set-types"
+import { registerDIHandler } from "../data-interactive-handler"
+import { DIHandler, DIResources, DIValues } from "../data-interactive-types"
+import {
+  attributeNotFoundResult, collectionNotFoundResult, dataContextNotFoundResult, errorResult,
+} from "./di-results"
+
+export const diCategoryColorMapHandler: DIHandler = {
+  create(resources: DIResources, _values?: DIValues & { metadata?: any }) {
+    const { dataContext, collection, attribute } = resources
+    if (!dataContext) return dataContextNotFoundResult
+    const metadata = getSharedCaseMetadataFromDataset(dataContext) ?? _values?.metadata
+    if (!metadata) return errorResult("Metadata not found")
+    if (!collection) return collectionNotFoundResult
+    const attributeId = attribute && typeof attribute === 'object' ? attribute.id : undefined
+    if (!attributeId) return attributeNotFoundResult
+    const createColorMap = (_values && 'colorMap' in _values) ? _values.colorMap as DICategoryColorMap : undefined
+    if (!createColorMap) return errorResult("New color map not found")
+    const categorySet = createProvisionalCategorySet(dataContext, attributeId)
+    metadata.promoteProvisionalCategorySet(categorySet)
+    if (!categorySet) return errorResult("Category set not found")
+    Object.entries(createColorMap || {}).forEach(([category, color]) => {
+      categorySet.setColorForCategory(category, color ?? categoryColorMap[category] ?? "")
+    })
+    categorySet.storeAllCurrentColors()
+    metadata.promoteProvisionalCategorySet(categorySet)
+    const categoryColorMap: Record<string, string | undefined> = categorySet?.colorMap || {}
+    if (!categoryColorMap) return errorResult("Category color map not found")
+
+    return {
+      success: true,
+      values: categoryColorMap
+    }
+
+  },
+
+  get(resources: DIResources, _values?: DIValues & { metadata?: any }) {
+    const { dataContext, collection, attribute } = resources
+    if (!dataContext) return dataContextNotFoundResult
+    if (!attribute) return attributeNotFoundResult
+    if (!collection) return collectionNotFoundResult
+
+    const metadata = getSharedCaseMetadataFromDataset(dataContext) ?? _values?.metadata
+    if (!metadata) return errorResult("Metadata not found")
+    const attributeId = attribute.id
+    const categorySet = metadata.getCategorySet(attributeId)
+    const categoryColorMap: Record<string, string | undefined> = categorySet?.colorMap || {}
+    if (!categoryColorMap) return errorResult("Category color map not found")
+
+    return {
+      success: true,
+      values: categoryColorMap
+    }
```

Comment:
Except follow the usual pattern for error results like `attributeNotFoundResult`, don't use a hard coded string here.
```suggestion
    if (!categorySet?.colorMap) return errorResult("Category color map not found")

    return {
      success: true,
      values: categorySet.colorMap
    }
```